### PR TITLE
fix(aws): raise ArgumentTypeError for parser

### PR DESCRIPTION
**Pull request by pedrooot**
### Description

It is needed to raise `ArgumentTypeError` when checking arguments for parser. Also, exceptions logic for custom messages is improved.

### Checklist

- Are there new checks included in this PR? Yes / No
    - If so, do we need to update permissions for the provider? Please review this carefully.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/providers/aws/aws_provider_test.py [hunk 1]
```diff
@@ -1370,7 +1369,7 @@ def test_test_connection_with_role_from_env_invalid_session_name(self):
         assert exception.type == AWSArgumentTypeValidationError
         assert (
             exception.value.args[0]
-            == "[1905] AWS argument type validation error - Role Session Name must be between 2 and 64 characters and may contain alphanumeric characters, periods, hyphens, and underscores."
+            == "[1905] Session Duration must be between 900 and 43200 seconds. - Role Session Name must be between 2 and 64 characters and may contain alphanumeric characters, periods, hyphens, and underscores."
```
Comment (jfagoagas) [inline tests/providers/aws/aws_provider_test.py]: This is strange... this exception should not refer to the session duration, could you take a look at it?

File: prowler/providers/aws/exceptions/exceptions.py [hunk 2]
```diff
@@ -8,188 +8,166 @@ class AWSBaseException(ProwlerException):
...
-            "remediation": "Check the AWS assume role configuration and ensure it is properly set up, please visithttps://docs.prowler.com/projects/prowler-open-source/en/latest/tutorials/aws/role-assumption/ and https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_terms-and-concepts.html#iam-term-role-session-name",
-            "file": "{file}",
-            "provider": "AWS",
+            "remediation": "Check the AWS assume role configuration and ensure it is properly set up, please visit https://docs.prowler.com/projects/prowler-open-source/en/latest/tutorials/aws/role-assumption/ and https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_terms-and-concepts.html#iam-term-role-session-name",
         },
     }
 
-    def __init__(self, code, provider="AWS", file=None, original_exception=None):
+    def __init__(
+        self, code, provider="AWS", file=None, original_exception=None, message=None
```
Comment (jfagoagas) [inline prowler/providers/aws/exceptions/exceptions.py]: Do you want to allow override the `provider`?

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/providers/aws/exceptions/exceptions.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/aws/exceptions/exceptions.py]: no, this is inside aws exceptions so provider will always be AWS

Comment (jfagoagas) [reply prowler/providers/aws/exceptions/exceptions.py]: In that case we can remove the argument and set it within.

File: tests/providers/aws/aws_provider_test.py (hunk 1 above)
Comment (pedrooot) [reply tests/providers/aws/aws_provider_test.py]: Yep, found an bug there

File: prowler/providers/aws/exceptions/exceptions.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/aws/exceptions/exceptions.py]: Done

**Review by jfagoagas — APPROVED** [review]
This is fantastic, thanks for this improvement and for fixing the little bugs with the CLI validation 😍

**Pull request merged**